### ISSUE TITLE
Automatic affinitization of worker threads to CPUs

### ISSUE THREAD

**Issue by craig-chasseur** [pivotalsoftware/quickstep#9]
To go with autodetection of the number of CPU cores in #7, it would be helpful to have a command-line flag that causes worker threads to be automatically pinned to individual CPU cores in a round-robin fashion.

**Comment by pateljm** [comment]
I agree! It's been on my mind too. But, I couldn't find anything that was part of C++ already. There is https://www.open-mpi.org/projects/hwloc/ from Open MPI, but seems like a lot to drag into here.

**Comment by craig-chasseur** [comment]
Nah, I can whip up something simpler than that just using regular Linux
syscalls (with fallback on no affinitization for other OSes).
